### PR TITLE
Assigned tasks grouped by assignee

### PR DESCRIPTION
**Pull request by styliii**
Sample response data
```
{
    "6994-b92e": [
        {
            "data": {
                "id": "da95-08d5",
                "type": "step",
                "attributes": {
                    "title": "Schedule a meeting with your Operations Guide to discuss",
                    "completed": false,
                    "kind": "Action",
                    "position": 4000,
                    "completedAt": null,
                    "description": null,
                    "decisionOptions": null,
                    "assigneeInfo": {
                        "id": "6994-b92e",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4"
                    }
                },
                "relationships": {
                    "process": {
                        "data": {
                            "id": "b031-3ddb",
                            "type": "workflowInstanceProcess"
                        }
                    },
                    "documents": {
                        "data": []
                    },
                    "selectedOption": {
                        "data": null
                    },
                    "assignee": {
                        "data": {
                            "id": "6994-b92e",
                            "type": "people"
                        }
                    }
                }
            },
            "included": [
                {
                    "id": "6994-b92e",
                    "type": "person",
                    "attributes": {
                        "email": "amos_bartell@feeney.io",
                        "firstName": "Winnifred",
                        "middleName": null,
                        "lastName": "Heller",
                        "phone": "990.446.9843",
                        "journeyState": null,
                        "personalEmail": null,
                        "about": null,
                        "primaryLanguage": null,
                        "updatedAt": "2023-01-25T16:52:26.468Z",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4",
                        "roleList": []
                    },
                    "relationships": {
                        "schools": {
                            "data": []
                        },
                        "schoolRelationships": {
                            "data": []
                        },
                        "address": {
                            "data": null
                        }
                    }
                }
            ]
        },
        {
            "data": {
                "id": "f1a6-499a",
                "type": "step",
                "attributes": {
                    "title": "Watch What is the Affiliation Agreement",
                    "completed": false,
                    "kind": "Action",
                    "position": 1000,
                    "completedAt": null,
                    "description": null,
                    "decisionOptions": null,
                    "assigneeInfo": {
                        "id": "6994-b92e",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4"
                    }
                },
                "relationships": {
                    "process": {
                        "data": {
                            "id": "74ff-13cb",
                            "type": "workflowInstanceProcess"
                        }
                    },
                    "documents": {
                        "data": [
                            {
                                "id": "6da4-15b7",
                                "type": "document"
                            }
                        ]
                    },
                    "selectedOption": {
                        "data": null
                    },
                    "assignee": {
                        "data": {
                            "id": "6994-b92e",
                            "type": "people"
                        }
                    }
                }
            },
            "included": [
                {
                    "id": "6da4-15b7",
                    "type": "document",
                    "attributes": {
                        "inheritanceType": null,
                        "title": "Video (G Drive)",
                        "link": "https://drive.google.com/file/d/1XSGTFVU6KCc3afJq19U9wxNrwtmGSLwN/view?usp=sharing",
                        "updatedAt": "2023-01-25T16:32:20.332Z"
                    }
                },
                {
                    "id": "6994-b92e",
                    "type": "person",
                    "attributes": {
                        "email": "amos_bartell@feeney.io",
                        "firstName": "Winnifred",
                        "middleName": null,
                        "lastName": "Heller",
                        "phone": "990.446.9843",
                        "journeyState": null,
                        "personalEmail": null,
                        "about": null,
                        "primaryLanguage": null,
                        "updatedAt": "2023-01-25T16:52:26.468Z",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4",
                        "roleList": []
                    },
                    "relationships": {
                        "schools": {
                            "data": []
                        },
                        "schoolRelationships": {
                            "data": []
                        },
                        "address": {
                            "data": null
                        }
                    }
                }
            ]
        },
        {
            "data": {
                "id": "ae6a-0bd9",
                "type": "step",
                "attributes": {
                    "title": "Watch Building a Wildflower Budget",
                    "completed": false,
                    "kind": "Action",
                    "position": 1000,
                    "completedAt": null,
                    "description": null,
                    "decisionOptions": null,
                    "assigneeInfo": {
                        "id": "6994-b92e",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4"
                    }
                },
                "relationships": {
                    "process": {
                        "data": {
                            "id": "0577-386d",
                            "type": "workflowInstanceProcess"
                        }
                    },
                    "documents": {
                        "data": [
                            {
                                "id": "497d-94df",
                                "type": "document"
                            }
                        ]
                    },
                    "selectedOption": {
                        "data": null
                    },
                    "assignee": {
                        "data": {
                            "id": "6994-b92e",
                            "type": "people"
                        }
                    }
                }
            },
            "included": [
                {
                    "id": "497d-94df",
                    "type": "document",
                    "attributes": {
                        "inheritanceType": null,
                        "title": "Video",
                        "link": "https://drive.google.com/file/d/13YwSSAJ-D9bVrGZIL9QCthSNhT6Ui7NM/view?usp=sharing",
                        "updatedAt": "2023-01-25T16:32:20.314Z"
                    }
                },
                {
                    "id": "6994-b92e",
                    "type": "person",
                    "attributes": {
                        "email": "amos_bartell@feeney.io",
                        "firstName": "Winnifred",
                        "middleName": null,
                        "lastName": "Heller",
                        "phone": "990.446.9843",
                        "journeyState": null,
                        "personalEmail": null,
                        "about": null,
                        "primaryLanguage": null,
                        "updatedAt": "2023-01-25T16:52:26.468Z",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4",
                        "roleList": []
                    },
                    "relationships": {
                        "schools": {
                            "data": []
                        },
                        "schoolRelationships": {
                            "data": []
                        },
                        "address": {
                            "data": null
                        }
                    }
                }
            ]
        },
        {
            "data": {
                "id": "e580-17f5",
                "type": "step",
                "attributes": {
                    "title": "Make a copy of the Visioning album template",
                    "completed": false,
                    "kind": "Action",
                    "position": 1000,
                    "completedAt": null,
                    "description": null,
                    "decisionOptions": null,
                    "assigneeInfo": {
                        "id": "6994-b92e",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4"
                    }
                },
                "relationships": {
                    "process": {
                        "data": {
                            "id": "b031-3ddb",
                            "type": "workflowInstanceProcess"
                        }
                    },
                    "documents": {
                        "data": [
                            {
                                "id": "ddf0-2d9b",
                                "type": "document"
                            }
                        ]
                    },
                    "selectedOption": {
                        "data": null
                    },
                    "assignee": {
                        "data": {
                            "id": "6994-b92e",
                            "type": "people"
                        }
                    }
                }
            },
            "included": [
                {
                    "id": "ddf0-2d9b",
                    "type": "document",
                    "attributes": {
                        "inheritanceType": null,
                        "title": "Google Doc",
                        "link": "https://docs.google.com/document/d/1mfkE-cUFAGwQi-W6ihy0PcXauTNb4aRWu1F4tIm20Cc",
                        "updatedAt": "2023-01-25T16:32:20.281Z"
                    }
                },
                {
                    "id": "6994-b92e",
                    "type": "person",
                    "attributes": {
                        "email": "amos_bartell@feeney.io",
                        "firstName": "Winnifred",
                        "middleName": null,
                        "lastName": "Heller",
                        "phone": "990.446.9843",
                        "journeyState": null,
                        "personalEmail": null,
                        "about": null,
                        "primaryLanguage": null,
                        "updatedAt": "2023-01-25T16:52:26.468Z",
                        "imageUrl": "https://avatars.githubusercontent.com/u/1683182?v=4",
                        "roleList": []
                    },
                    "relationships": {
                        "schools": {
                            "data": []
                        },
                        "schoolRelationships": {
                            "data": []
                        },
                        "address": {
                            "data": null
                        }
                    }
                }
            ]
        }
    ]
}
```

### CODE REVIEW

**Review by keithtom — APPROVED** [review]
Seems good to me minus that 1 extra file

File: app/controllers/v1/ssj/dashboard_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    workflow = Workflow::Instance::Workflow.find_by!(external_identifier: params[:workflow_id])
+    process_ids = Workflow::Instance::Process.where(workflow_id: workflow.id).pluck(:id)
+    steps = Workflow::Instance::Step.where(process_id: process_ids, completed: false).where.not(assignee_id: nil).
+      includes(:process, :documents, :selected_option, :assignee, definition: [:documents]).
+      group_by{|step| step.assignee.external_identifier }
+
+    options = {}
+    options[:include] = ['documents', 'assignee']
+    render json: V1::Ssj::AssignedStepsSerializer.new(steps, options)
+  end
```
Comment (keithtom) [inline app/controllers/v1/ssj/dashboard_controller.rb:13]: I just realized by looking at the response, the assignee is serialized multiple times...  that's a lot of data overhead.  not sure how to achieve this but the assignee "include" should only be on the top level object... not each step if that makes sense...

Comment (keithtom) [reply app/controllers/v1/ssj/dashboard_controller.rb:13]: Also looking at the person serializer, we should find a way to pass in an option that's like `params[:basic] = true` or something like that that just has what Taylor needs to put a name, email and icon.

Comment (keithtom) [reply app/controllers/v1/ssj/dashboard_controller.rb:13]: Maybe for these one off SSJ controllers that arent following JSON API specs anyway, we just construct our own json without going through the serializer - it doesn't seem like it is helping that much anyway... more of an obstacle.  We can still use them to serialize objects but don't need to build the whole response with the assigned steps serializer?  Just a thought.

**Review by keithtom — CHANGES_REQUESTED** [review]
Put a few comments.  I think we should re-think the approach.  Take what we like from jsonapi but don't force us to have to use it to render the whole response...  ideally we still have the `data` and `included` keys, like try to follow it where it makes sense...

File: app/controllers/v1/ssj/dashboard_controller.rb (hunk 1 above)
Comment (styliii) [reply app/controllers/v1/ssj/dashboard_controller.rb:13]: yup this makes sense to me. i agree that we should strip out unneeded data. There isn't a way to pass params to a child serializer via fast json api serializer. so yeah, will try to look into using our own.

**Review by keithtom — APPROVED** [review]
Seems good to me.  I forgot why in `step_serializer` we have `belongs_to :assignee` and `attribute :assignee_info`.   I think that was previous hack that we can cleanup another time.

**Comment by styliii** [comment]
> Seems good to me. I forgot why in `step_serializer` we have `belongs_to :assignee` and `attribute :assignee_info`. I think that was previous hack that we can cleanup another time.

Yes, it was a hack that made it easier for the FE to get the assignee info, instead of looking in `included`, i think.

**Pull request merged**